MetaMask/metamask-extension
# Fix fallback gas estimation

## Explanation

Our fallback gas estimation was failing due to a bug in the `@metamask/controller-utils` package. This was causing gas estimation to fail completely on certain networks (those not supported by our gas estimation APIs and non EIP-1559 compatibile), and it was causing the fallback gas estimation operation (in case our API was down) to fail across all networks.

Additionally, the legacy gas API has been disabled as part of this PR. This was required in order to cover this functionality with e2e tests (if the legacy gas API was enabled, that would take precedence over the code this PR is fixing).

I tried removing the legacy gas API in a separate PR, but that depends upon this bug fix, so they need to be changed together.

A few test IDs were added for the purpose of helping with writing the e2e tests.

Fixes https://github.com/MetaMask/metamask-extension/issues/19735

## Manual Testing Steps

Switch to the BNB network, and start the send or swaps flow. You should see an error like the one shown in the referenced issue if you're testing with `develop` or the v10.33.0 release candidate. With this PR, the error no longer appears.

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
